File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/VideoApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class VideoApi {
    private final HttpClient client;
    
    public VideoApi(HttpClient client) {
        this.client = client;
    }

    /** List videos */
    public OpenAiVideoList listVideos(Map<String, Object> params) throws Exception {
        return (OpenAiVideoList) client.get(ApiPaths.aiPath("/videos"), params);
    }

    /** Create video */
    public OpenAiVideo createVideo(VideoGenerationRequest body) throws Exception {
        return (OpenAiVideo) client.post(ApiPaths.aiPath("/videos"), body);
    }

    /** Create video */
    public OpenAiVideo createVideoGenerations(VideoGenerationRequest body) throws Exception {
        return (OpenAiVideo) client.post(ApiPaths.aiPath("/videos/generations"), body);
    }

    /** Create video */
    public OpenAiVideo createVideoGenerations2(VideoGenerationRequest body) throws Exception {
        return (OpenAiVideo) client.post(ApiPaths.aiPath("/v1/videos/generations"), body);
    }

    /** Retrieve video */
    public OpenAiVideo retrieve(String video_id) throws Exception {
        return (OpenAiVideo) client.get(ApiPaths.aiPath("/videos/" + video_id + ""));
    }

    /** Delete video */
    public OpenAiVideoDeleteResponse deleteVideo(String video_id) throws Exception {
        return (OpenAiVideoDeleteResponse) client.delete(ApiPaths.aiPath("/videos/" + video_id + ""));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/VideoApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class VideoApi(private val client: HttpClient) {

    /** List videos */
    suspend fun listVideos(params: Map<String, Any>? = null): OpenAiVideoList? {
        return client.get(ApiPaths.aiPath("/videos"), params) as? OpenAiVideoList
    }

    /** Create video */
    suspend fun createVideo(body: VideoGenerationRequest): OpenAiVideo? {
        return client.post(ApiPaths.aiPath("/videos"), body) as? OpenAiVideo
    }

    /** Create video */
    suspend fun createVideoGenerations(body: VideoGenerationRequest): OpenAiVideo? {
        return client.post(ApiPaths.aiPath("/videos/generations"), body) as? OpenAiVideo
    }

    /** Create video */
    suspend fun createVideoGenerations2(body: VideoGenerationRequest): OpenAiVideo? {
        return client.post(ApiPaths.aiPath("/v1/videos/generations"), body) as? OpenAiVideo
    }

    /** Retrieve video */
    suspend fun retrieve(video_id: String): OpenAiVideo? {
        return client.get(ApiPaths.aiPath("/videos/$video_id")) as? OpenAiVideo
    }

    /** Delete video */
    suspend fun deleteVideo(video_id: String): OpenAiVideoDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/videos/$video_id")) as? OpenAiVideoDeleteResponse
    }
}
